tasks.py

```diff
@@ -248,8 +260,36 @@ def deploy_core(ctx, name, version=None, no_tests=False):
 
 
 @task
-def deploy_lambda_package(ctx, name):
-    aws_lambda.deploy(os.getcwd(), local_package='../..', requirements='../../requirements.txt')
+def deploy_lambda_package(ctx, name, suffix):
+    # create the temporary local dev lambda directories
+    if suffix:
+        new_name = name + '_' + suffix
+        new_src = '../' + new_name
+        cmd_mkdir = "rm -fr %s; mkdir -p %s" % (new_src, new_src)
+        cmd_copy = "cp -r * %s" % new_src
+        cmd_cd = "cd %s" % new_src
+        cmd_modify_cfg = "sed 's/%s/%s/g' config.yaml > config.yaml#" % (name, new_name)
+        cmd_replace_cfg = "mv config.yaml# config.yaml"
+        cmd = ';'.join([cmd_mkdir, cmd_copy, cmd_cd, cmd_modify_cfg, cmd_replace_cfg])
+        print(cmd)
+        run(cmd)
+    else:
+        new_name = name
+    with chdir(new_src):
+        aws_lambda.deploy(os.getcwd(), local_package='../..', requirements='../../requirements.txt')
+    # add roles and environment variables
+    role_arn = 'arn:aws:iam::643366669028:role/lambda_full_s3'  # temporarily hardcorded
+    lambda_update_config = {'FunctionName': new_name, 'Role': role_arn}
```

Comment:
for env variables... also aws_lambda will set those as part of the deploy.. the just need to start with LAMBDA_ see here: https://github.com/j1z0/python-lambda/blob/master/aws_lambda/aws_lambda.py#L356

Reply:
so I think you can remove the whole update_function call and lets aws_lambda do it for you.

Reply:
Thanks Jeremy. Good to know. But this way we can control different env variables for different lambdas. Not sure how useful that might be though..

Reply:
Sorry I just tried it and it didn't work. Let's keep it this way for now and move on. We can come back later if we wanted to.